tests/test_integration.py

```diff
@@ -1338,3 +1340,41 @@ def test_issues_539_abi3_resolution():
     res.assert_success()
 
     subprocess.check_call([cryptography_pex, '-c', 'import cryptography'])
+
+
+@pytest.mark.skip("Acceptance test for landing https://github.com/pantsbuild/pex/issues/716.")
+def test_reproducible_build():
+
+  def assert_reproducible(args):
+    with temporary_dir() as td:
+      pex1 = os.path.join(td, '1.pex')
+      pex2 = os.path.join(td, '2.pex')
+      run_pex_command(args + ['-o', pex1])
```

Comment:
So, you may miss alot unless you set two different PYTHONHASHSEED. Pytest sets one via ... I'm not sure. But if via env car export, you may get that inherited for both PEX commands which is not what you want to ferret out the datastructure-related issues.

Reply:
Agreed. Good catch. Setting the `env` in `run_pex_command` will temporarily override for the subprocess whatever Tox sets beforehand (which indeed is via an env var https://tox.readthedocs.io/en/latest/example/basic.html#special-handling-of-pythonhashseed).